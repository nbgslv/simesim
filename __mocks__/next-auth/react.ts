const mockSession = {
  expires: new Date(Date.now() + 2 * 86400).toISOString(),
  user: {
    email: '0512345678',
    emailEmail: 'test@test.com',
    id: 'clcclcclcclcclcclc1234',
    image: null,
    name: 'foo bar',
    role: 'ADMIN',
  },
};

export const useSession = jest.fn(
  () => ({ data: mockSession, status: 'authenticated' }) // return type is [] in v3 but changed to {} in v4
);
export const signOut = jest.fn(() =>
  Promise.resolve({ url: process.env.NEXTAUTH_URL as string })
);
