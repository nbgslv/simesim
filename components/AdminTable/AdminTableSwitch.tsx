import React from 'react';
import { Spinner, Form } from 'react-bootstrap';
import { GridRowId, GridRowModel, GridValidRowModel } from '@mui/x-data-grid';

type AdminTableSwitchProps<T extends GridValidRowModel> = {
  checked: boolean;
  onChange: (checked: boolean, rowId: GridRowId, row: GridRowModel<T>) => void;
  rowId: GridRowId;
  row: GridRowModel<T>;
  loading: GridRowId;
};

const AdminTableSwitch = <T extends GridValidRowModel>({
  checked,
  onChange,
  rowId,
  row,
  loading,
}: AdminTableSwitchProps<T>) => {
  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
    onChange(event.target.checked, rowId, row);
  };

  if (loading === rowId)
    return <Spinner className="mx-auto" animation={'border'} />;
  return <Form.Check type="switch" checked={checked} onChange={handleChange} />;
};

export default AdminTableSwitch;
