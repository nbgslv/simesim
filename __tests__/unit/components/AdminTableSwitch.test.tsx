import { fireEvent, render, screen } from '@testing-library/react';
import AdminTableSwitch from '../../../components/AdminTable/AdminTableSwitch';

const handeOnChange = jest.fn();

describe('AdminTableSwitch', () => {
  it('renders the switch', () => {
    const { container } = render(
      <AdminTableSwitch
        checked={false}
        onChange={handeOnChange}
        rowId={'1'}
        row={{ id: 1, name: 'John Smith', age: 35 }}
        loading={''}
      />
    );
    expect(
      container.querySelector('input[type="checkbox"]')
    ).toBeInTheDocument();
  });
  it('renders the spinner when loading', () => {
    const { container } = render(
      <AdminTableSwitch
        checked={false}
        onChange={handeOnChange}
        rowId={'1'}
        row={{ id: 1, name: 'John Smith', age: 35 }}
        loading={'1'}
      />
    );

    expect(container.querySelector('.spinner-border')).toBeInTheDocument();
  });
  it('calls onChange when the switch is clicked', () => {
    render(
      <AdminTableSwitch
        checked={false}
        onChange={handeOnChange}
        rowId={'1'}
        row={{ id: 1, name: 'John Smith', age: 35 }}
        loading={''}
      />
    );
    fireEvent.click(screen.getByRole('checkbox'));
    expect(handeOnChange).toHaveBeenCalled();
  });
});
