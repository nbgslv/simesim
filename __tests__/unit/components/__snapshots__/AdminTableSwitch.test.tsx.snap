// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AdminTableSwitch matches snapshot 1`] = `
<div>
  <div
    class="form-switch"
  >
    <input
      class="form-check-input"
      type="checkbox"
    />
  </div>
</div>
`;
